tests/integration/kernel/src/main.c

```diff
@@ -0,0 +1,305 @@
+
+	/* Get items from lifo */
+	for (i = 0U; i < LIST_LEN; i++) {
+		rx_data = k_lifo_get(&lifo, K_NO_WAIT);
+		zassert_equal(rx_data->value, lifo_data[LIST_LEN-1-i].value);
+	}
+
+	/* Clear the spawn thread */
+	k_thread_abort(tid);
+
+}
+
+/* STACK SCENARIO */
+/* Thread A enters items into a stack, starts thread B and waits for a semaphore. */
+/* Thread B extracts all items from the stack and enters some items back into the stack. */
+/* Thread B gives the semaphore for Thread A to continue. */
+/* Once the control is returned back to Thread A, it extracts all items from the stack. */
+/* Verify the data's correctness. */
+
+
+#define STACK_LEN	8
+#define MAX_ITEMS	8
+
+K_STACK_DEFINE(stack, STACK_LEN);
+stack_data_t stack_data[MAX_ITEMS];
+
+
+static void thread_entry_fn_stack(void *p1, void *p2, void *p3)
+{
+	stack_data_t *rx_data;
+	stack_data_t data[MAX_ITEMS];
+
+	/* fill data to compare */
+	for (int i = 0; i < STACK_LEN; i++) {
+		data[i] = i;
+	}
+
+	/* read data from a stack */
+	for (int i = 0; i < STACK_LEN; i++) {
+		k_stack_pop((struct k_stack *)p1, (stack_data_t *)&rx_data, K_NO_WAIT);
+	}
+
+	zassert_false(memcmp(rx_data, data, STACK_LEN),
+		      "Push & Pop items does not match");
+
+	/* Push data into a stack */
+	for (int i = 0; i < STACK_LEN; i++) {
+		stack_data[i] *= i;
+		k_stack_push((struct k_stack *)p1, (stack_data_t)&stack_data[i]);
+	}
+
+	/* Give control back to Thread A */
+	k_sem_give(&sema);
+}
+
+
+ZTEST(kernel, test_stack_usage)
+{
+	stack_data_t *rx_data;
+
+	/* Init binary semaphore */
+	k_sem_init(&sema, 0, 1);
+
+	/* Push data into a stack */
+	for (int i = 0; i < STACK_LEN; i++) {
+		stack_data[i] = i;
+		k_stack_push(&stack, (stack_data_t)&stack_data[i]);
+	}
+
+	/* Create the Thread B */
+	k_tid_t tid = k_thread_create(&tdata, tstack, STACK_SIZE,
+					thread_entry_fn_stack, &stack, NULL, NULL,
+					K_PRIO_PREEMPT(0), K_INHERIT_PERMS, K_NO_WAIT);
+
+	/* Let the thread B run */
+	k_sem_take(&sema, K_FOREVER);
+
+	/* read data from a stack */
+	for (int i = 0; i < STACK_LEN; i++) {
+		k_stack_pop(&stack, (stack_data_t *)&rx_data, K_NO_WAIT);
+	}
+
+
+	/* Verify the correctness */
+	zassert_false(memcmp(rx_data, stack_data, STACK_LEN),
+		      "Push & Pop items does not match");
+
+	/* Clear the spawn thread */
+	k_thread_abort(tid);
+}
+
+/* MUTEX SCENARIO */
+/* Initialize the mutex. */
+/* Start the two Threads with the entry points. */
+/* The entry points change variable and use lock and unlock mutex functions to */
+/* synchronize the access to that variable. */
+/* Join all threads. */
+/* Verify the variable. */
+
+#define NUMBER_OF_ITERATIONS 1000
+
+static uint32_t mutex_data;
+static struct k_thread tdata_2;
+static K_THREAD_STACK_DEFINE(tstack_2, STACK_SIZE);
+
+K_MUTEX_DEFINE(mutex);
+
+static void thread_entry_fn_mutex(void *p1, void *p2, void *p3)
+{
+	uint32_t i;
+
+	k_mutex_lock((struct k_mutex *)p1, K_FOREVER);
+	for (i = 0; i < NUMBER_OF_ITERATIONS; i++) {
+		mutex_data += 1;
+	}
```

Comment:
Loop seems unnecessary - why not just add it directly: `mutex_data += NUMBER_OF_ITERATIONS`? Either way we do not unlock the value between additions.

Reply:
I'd like to simulate long time processes instead of one line increasing the value of mutex_data.